src/main/java/com/atlassian/db/replica/api/reason/RouteDecision.java

```diff
@@ -38,14 +42,31 @@ public Reason getReason() {
         return Optional.ofNullable(cause);
     }
 
+    /**
+     * @return information whether reason was caused by write sql operation.
+     */
+    public boolean isWrite(){
+        return WRITE_OPERATION.equals(reason) ||
+                LOCK.equals(reason) ||
+                RW_API_CALL.equals(reason);
+    }
+
+    /**
+     * @return whether sql was run on Main
+     */
+    public boolean isRunOnMain() {
```

Comment:
Any reason to add this?

Reply:
It's semantic. The main point of the class is to *decide* to go to main or not. From API it seems that `RouteDecision` has a "Reason", but doesn't even know what the decision is.
If `isWrite` will go here, then it would also add symmetry.

Reply:
I don't have strong opinion here, it just looks a bit nicer to have all of this inside RouteDecision.
I've failed adding `write` field into `Reason`. So this looks to be nicer.

Reply:
So we are moving it from 'Reason`? I would like to make sure we don't create two alternative ways to reach the same data. The reason to mix Reason and `isWrite` in one object was to not allow broken states like `RW_OPeration' isWrite=false on compile time.

Reply:
Good catch!

- [ if isRunOnMain() moved, delete it from `Reason`]

Reply:
If `Reason.isRunOnMain` becomes package-private, then we get both the internal consistency and one way to call the API.

Reply:
I'll prefer to migrate internals to also use  `RouteDecision.isRunOnMain()` and have 1 source of truth